README.md
# spreed-webrtc Debian packaging

## Example build steps
First, grab the packaging from Git and configure your checkout. Omit this if you
already have a checkout of the packaging.

    $ git clone git@git.intranet.struktur.de:debian/spreed-webrtc
    $ cd spreed-webrtc
    $ git checkout upstream
    $ git checkout pristine-tar
    $ git checkout ubuntu/precise

If you're releasing a new upstream version, you should start by updating and
committing a Debian changelog entry for the new upstream version $VERSION.

    $ dch -v $VERSION-1

Next, you should pull in the new upstream tarball as follows:

    $ ./debian/rules get-orig-source
    $ mv spreed-webrtc_$VERSION.orig.tar.xz ..
    $ git-import-orig ../spreed-webrtc_$VERSION.orig.tar.xz
    What is the upstream version? [$VERSION]
    <hit enter, or provide -u$VERSION flag to bypass>

At this point, you should have successfully imported the new upstream tarball.

If you're working on releasing a packaging change for someone else's upstream
version, you'll need to obtain the original source tarball for the upstream
$VERSION as follows:

    $ pristine-tar checkout spreed-webrtc_$VERSION.orig.tar.xz
    $ mv spreed-webrtc_$VERSION.orig.tar.xz ..

Assuming you have a pbuilder $BASETGZ (typically stored under
`/var/cache/pbuilder`) for the given distribution prepared, you may now proceed
to build the package as follows:

    $ # You may skip updating if the pbuilder is known to be up to date.
    $ sudo pbuilder update --basetgz $BASETGZ
    $ pdebuild --buildresult `pwd`/../ --debbuildopts -sa -- --basetgz $BASETGZ

All going well, you should have a binary package in the directory specified by
`--buildresult`. Use `-S` as needed to produce a source build for uploading to
Launchpad. Then do the upload as follows:

    $ debsign ../spreed-webrtc_$VERSION-1_source.changes 
    $ dput ppa:strukturag/spreed-webrtc ../spreed-webrtc_$VERSION-1_source.changes

Once any archive uploads have been accepted, tag the version and push your
changes:

    $ git tag ubuntu/precise/$VERSION-1
    $ git push --all
    $ git push --tags



debian/changelog
spreed-webrtc (0.29.7-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.7

 -- Joachim Bauch <bauch@struktur.de>  Wed, 16 Aug 2017 08:53:28 +0200

spreed-webrtc (0.29.6-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.6
  * Upgrade to pkac 0.1.0

 -- Joachim Bauch <bauch@struktur.de>  Fri, 05 May 2017 10:39:22 +0200

spreed-webrtc (0.29.5-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.5

 -- Joachim Bauch <bauch@struktur.de>  Fri, 02 Dec 2016 16:58:45 +0100

spreed-webrtc (0.29.4-1~trusty2) trusty; urgency=medium

  * Re-upload to force Launchpad package lists update.

 -- Joachim Bauch <bauch@struktur.de>  Fri, 25 Nov 2016 15:26:05 +0100

spreed-webrtc (0.29.4-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.4

 -- Joachim Bauch <bauch@struktur.de>  Thu, 24 Nov 2016 11:34:51 +0100

spreed-webrtc (0.29.3-1~trusty2) trusty; urgency=medium

  * Rebuild against updated golang-github-strukturag-spreed-
    turnservicecli.

 -- Joachim Bauch <bauch@struktur.de>  Mon, 21 Nov 2016 13:58:47 +0100

spreed-webrtc (0.29.3-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.3

 -- Joachim Bauch <bauch@struktur.de>  Thu, 10 Nov 2016 11:55:28 +0100

spreed-webrtc (0.29.2-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.2
  * Bump minimal phoenix to 1.0.2

 -- Simon Eisenmann <simon@struktur.de>  Thu, 13 Oct 2016 13:56:53 +0200

spreed-webrtc (0.29.1-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.1
  * Remove obsolete patch

 -- Simon Eisenmann <simon@struktur.de>  Wed, 07 Sep 2016 16:04:31 +0200

spreed-webrtc (0.29.0-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.29.0
  * Add turnservicecli build dependency
  * Fixup linter patch

 -- Simon Eisenmann <simon@struktur.de>  Tue, 30 Aug 2016 09:46:56 +0200

spreed-webrtc (0.28.0-1~trusty2) trusty; urgency=medium

  * Add patch to fix compilation with Go 1.4.

 -- Joachim Bauch <bauch@struktur.de>  Fri, 12 Aug 2016 16:59:58 +0200

spreed-webrtc (0.28.0-1~trusty1) trusty; urgency=medium

  * Imported Upstream version 0.28.0
  * Updated dependencies to "golang-go (>= 1.4)".

 -- Joachim Bauch <bauch@struktur.de>  Fri, 12 Aug 2016 16:49:27 +0200

spreed-webrtc (0.27.1-2~trusty1) trusty; urgency=medium

  * Updated go.uuid dependency

 -- Simon Eisenmann <simon@struktur.de>  Wed, 03 Aug 2016 14:47:55 +0200

spreed-webrtc (0.27.1-1~trusty2) trusty; urgency=medium

  * Add armhf and arm64 builds

 -- Simon Eisenmann <simon@struktur.de>  Wed, 27 Jul 2016 09:27:23 +0200

spreed-webrtc (0.27.1-1~trusty1) trusty; urgency=low

  [ Joachim Bauch ]
  * Add support for systemd.

  [ Simon Eisenmann ]

 -- Joachim Bauch <bauch@struktur.de>  Fri, 22 Jul 2016 13:53:39 +0200

spreed-webrtc (0.27.1-1~trusty0) trusty; urgency=medium

  * Imported Upstream version 0.27.1

 -- Simon Eisenmann <simon@struktur.de>  Mon, 11 Jul 2016 18:14:49 +0200

spreed-webrtc (0.27.0-1~trusty1) trusty; urgency=medium

  * Add support for triggers

 -- Simon Eisenmann <simon@struktur.de>  Mon, 11 Jul 2016 13:50:31 +0200

spreed-webrtc (0.27.0-1~trusty0) trusty; urgency=low

  * Imported Upstream version 0.27.0

 -- Joachim Bauch <bauch@struktur.de>  Fri, 08 Jul 2016 11:03:58 +0200

spreed-webrtc (0.26.0-1~trusty0) trusty; urgency=low

  * Imported Upstream version 0.26.0
  * Use new release branches.
  * Updated branches.

 -- Joachim Bauch <bauch@struktur.de>  Fri, 08 Jul 2016 10:51:34 +0200

spreed-webrtc (0.25.5-1~trusty0) trusty; urgency=medium

  * Imported Upstream version 0.25.5

 -- Simon Eisenmann <simon@struktur.de>  Fri, 06 May 2016 14:16:38 +0200

spreed-webrtc (0.25.4-1~trusty0) trusty; urgency=medium

  * Imported Upstream version 0.25.4
  * Remove systemgo patch which is now upstream.

 -- Simon Eisenmann <simon@struktur.de>  Wed, 13 Apr 2016 15:51:23 +0200

spreed-webrtc (0.25.3-1~trusty0) trusty; urgency=medium

  * Cleanup obsolete init scripting.
  * Imported Upstream version 0.25.3
  * Updated for Golang modules packaging.

 -- Simon Eisenmann <simon@struktur.de>  Wed, 30 Mar 2016 10:53:38 +0200

spreed-webrtc (0.25.1-1~trusty2) trusty; urgency=low

  * New arch-indep packages replace old single package.

 -- Joachim Bauch <bauch@struktur.de>  Thu, 03 Mar 2016 12:40:53 +0100

spreed-webrtc (0.25.1-1~trusty1) trusty; urgency=low

  * Next version will require golang-phoenix-dev 1.0.0 or newer.
  * Imported Upstream version 0.25.1

 -- Joachim Bauch <bauch@struktur.de>  Wed, 02 Mar 2016 14:52:42 +0100

spreed-webrtc (0.25.0-1~trusty2) trusty; urgency=low

  * Fixed typo.
  * Moved binaries to separate package.

 -- Joachim Bauch <bauch@struktur.de>  Wed, 02 Mar 2016 12:08:20 +0100

spreed-webrtc (0.25.0-1~trusty1) trusty; urgency=low

  * One dependency per line, wrap description text.
  * Version 0.25.x requires Golang 1.3 or newer.
  * Updated gbp configuration for new branch.
  * Next version needs "golang-nats-dev".
  * Split static assets into separate package to avoid having to rebuild on all architectures.
  * Also install "CHANGELOG.md".
  * Imported Upstream version 0.25.0

 -- Joachim Bauch <bauch@struktur.de>  Wed, 02 Mar 2016 11:35:44 +0100

spreed-webrtc (0.24.12-1~trusty1) trusty; urgency=low

  * Imported Upstream version 0.24.12
  * Changes in 0.24.12
   + Brought back mediaDevices wrapper for gUM for Firefox >= 38 fixing #263 and #264.

 -- Joachim Bauch <bauch@struktur.de>  Tue, 01 Mar 2016 17:55:59 +0100

spreed-webrtc (0.24.11-1~trusty2) trusty; urgency=low

  * Rotate logfile.

 -- Joachim Bauch <bauch@struktur.de>  Tue, 23 Feb 2016 13:03:13 +0100

spreed-webrtc (0.24.11-1~trusty1) trusty; urgency=low

  * Imported Upstream version 0.24.11
  * Changes in 0.24.11
   + Stop waiting on video early if first video track is enabled but muted.
   + Use sh shebang instead of bash to be less Linux specific (#244).
   + Updated deps to no longer user code.google.com.

debian/spreed-webrtc.init.d
#!/bin/sh
### BEGIN INIT INFO
# Provides:          spreed-webrtc
# Required-Start:    $network $local_fs
# Required-Stop:
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
# Short-Description: Spreed Speak Freely server
# Description:       WebRTC video conferencing service.
### END INIT INFO

# Author: Lance Cooper <lance@struktur.de>

PATH=/sbin:/usr/sbin:/bin:/usr/bin
DESC='Spreed Speak Freely'
NAME=spreed-webrtc
DAEMON=/usr/sbin/spreed-webrtc-server
SCRIPTNAME=/etc/init.d/$NAME

# Bail out if we're running under Upstart.
if [ "$1" = start ] && which initctl >/dev/null && initctl version | grep -q upstart; then
    exit 1
fi

# Exit if the package is not installed
[ -x $DAEMON ] || exit 0

# Read defaults.
if [ -r /etc/default/$NAME ]; then
    source /etc/default/$NAME
else
    exit 0
fi

# Load the VERBOSE setting and other rcS variables
. /lib/init/vars.sh

# Define LSB log_* functions.
# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
. /lib/lsb/init-functions

#
# Function that starts the daemon/service
#
do_start()
{
	start-stop-daemon --start \
                      --quiet \
                      --pidfile $WEBRTC_PID \
                      --startas $DAEMON \
                      --test > /dev/null \
	|| return 1

    # Create the run directory.
    test -e $WEBRTC_RUN_DIR || mkdir -p $WEBRTC_RUN_DIR || true
    chown -R $WEBRTC_USER:$WEBRTC_GROUP $WEBRTC_RUN_DIR || true
    chmod 770 $WEBRTC_RUN_DIR || true

    # Set some performance parameters
    ulimit -n $WEBRTC_NOFILE
    export GOMAXPROCS=$WEBRTC_GOMAXPROCS

    start-stop-daemon --start \
                      --quiet \
                      --make-pidfile \
                      --pidfile $WEBRTC_PID \
                      --chuid $WEBRTC_USER \
                      --group $WEBRTC_GROUP \
                      --startas $DAEMON \
                      -- \
                      -c $WEBRTC_CONF \
                      -l $WEBRTC_LOG \
                      $WEBRTC_ARGS \
    || return 2

    return 0
}

#
# Function that stops the daemon/service
#
do_stop()
{
	start-stop-daemon --stop \
                      --quiet \
                      --retry=TERM/30/KILL/5 \
                      --pidfile $WEBRTC_PID \
                      --name $NAME
	RETVAL="$?"
	[ "$RETVAL" = 2 ] && return 2

	rm -f $WEBRTC_PID
	return "$RETVAL"
}

#
# Function that sends a SIGHUP to the daemon/service
#
do_reload() {
    do_stop
    do_start
	return 0
}

case "$1" in
  start)
    [ "$VERBOSE" != no ] && log_daemon_msg "Starting $DESC " "$NAME"
    do_start
    case "$?" in
		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
	esac
  ;;
  stop)
	[ "$VERBOSE" != no ] && log_daemon_msg "Stopping $DESC" "$NAME"
	do_stop
	case "$?" in
		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
	esac
	;;
  status)
       status_of_proc "$DAEMON" "$NAME" && exit 0 || exit $?
       ;;
  restart|force-reload)
	log_daemon_msg "Restarting $DESC" "$NAME"
	do_stop
	case "$?" in
	  0|1)
		do_start
		case "$?" in
			0) log_end_msg 0 ;;
			1) log_end_msg 1 ;; # Old process is still running
			*) log_end_msg 1 ;; # Failed to start
		esac
		;;
	  *)
	  	# Failed to stop
		log_end_msg 1
		;;
	esac
	;;
  *)
	echo "Usage: $SCRIPTNAME {start|stop|status|restart|force-reload}" >&2
	exit 3
	;;
esac

:


static/js/filters/filters.js
/*
 * Spreed Speak Freely.
 * Copyright (C) 2013-2014 struktur AG
 *
 * This file is part of Spreed Speak Freely.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
define([
  'underscore',

  'filters/displayname',
  'filters/buddyimagesrc',
  'filters/displayconference'
], function(_, displayName, buddyImageSrc, displayConference) {

  var filters = {
      displayName: displayName,
      buddyImageSrc: buddyImageSrc,
      displayConference: displayConference
  };

  var initialize = function (angModule) {
    _.each(filters, function(filter, name) {
      angModule.filter(name, filter);
    })
  }

  return {
    initialize: initialize
  };

});

debian/spreed-speakfreely.init.d
#!/bin/sh
### BEGIN INIT INFO
# Provides:          spreed-speakfreely
# Required-Start:    $network $local_fs
# Required-Stop:
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
# Short-Description: Spreed Speak Freely server
# Description:       WebRTC video conferencing service.
### END INIT INFO

# Author: Lance Cooper <lance@struktur.de>

PATH=/sbin:/usr/sbin:/bin:/usr/bin
DESC='Spreed Speak Freely'
NAME=spreed-speakfreely
DAEMON=/usr/sbin/spreed-speakfreely-server
SCRIPTNAME=/etc/init.d/$NAME

# Bail out if we're running under Upstart.
if [ "$1" = start ] && which initctl >/dev/null && initctl version | grep -q upstart; then
    exit 1
fi

# Exit if the package is not installed
[ -x $DAEMON ] || exit 0

# Read defaults.
if [ -r /etc/default/$NAME ]; then
    source /etc/default/$NAME
else
    exit 0
fi

# Load the VERBOSE setting and other rcS variables
. /lib/init/vars.sh

# Define LSB log_* functions.
# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
. /lib/lsb/init-functions

#
# Function that starts the daemon/service
#
do_start()
{
	start-stop-daemon --start \
                      --quiet \
                      --pidfile $SPEAKFREELY_PID \
                      --startas $DAEMON \
                      --test > /dev/null \
	|| return 1

    # Create the run directory.
    test -e $SPEAKFREELY_RUN_DIR || mkdir -p $SPEAKFREELY_RUN_DIR || true
    chown -R $SPEAKFREELY_USER:$SPEAKFREELY_GROUP $SPEAKFREELY_RUN_DIR || true
    chmod 770 $SPEAKFREELY_RUN_DIR || true

    # Set some performance parameters
    ulimit -n $SPEAKFREELY_NOFILE
    export GOMAXPROCS=$SPEAKFREELY_GOMAXPROCS

    start-stop-daemon --start \
                      --quiet \
                      --make-pidfile \
                      --pidfile $SPEAKFREELY_PID \
                      --chuid $SPEAKFREELY_USER \
                      --group $SPEAKFREELY_GROUP \
                      --startas $DAEMON \
                      -- \
                      -c $SPEAKFREELY_CONF \
                      -l $SPEAKFREELY_LOG \
                      $SPEAKFREELY_ARGS \
    || return 2

    return 0
}

#
# Function that stops the daemon/service
#
do_stop()
{
	start-stop-daemon --stop \
                      --quiet \
                      --retry=TERM/30/KILL/5 \
                      --pidfile $SPEAKFREELY_PID \
                      --name $NAME
	RETVAL="$?"
	[ "$RETVAL" = 2 ] && return 2

	rm -f $SPEAKFREELY_PID
	return "$RETVAL"
}

#
# Function that sends a SIGHUP to the daemon/service
#
do_reload() {
    do_stop
    do_start
	return 0
}

case "$1" in
  start)
    [ "$VERBOSE" != no ] && log_daemon_msg "Starting $DESC " "$NAME"
    do_start
    case "$?" in
		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
	esac
  ;;
  stop)
	[ "$VERBOSE" != no ] && log_daemon_msg "Stopping $DESC" "$NAME"
	do_stop
	case "$?" in
		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
	esac
	;;
  status)
       status_of_proc "$DAEMON" "$NAME" && exit 0 || exit $?
       ;;
  restart|force-reload)
	log_daemon_msg "Restarting $DESC" "$NAME"
	do_stop
	case "$?" in
	  0|1)
		do_start
		case "$?" in
			0) log_end_msg 0 ;;
			1) log_end_msg 1 ;; # Old process is still running
			*) log_end_msg 1 ;; # Failed to start
		esac
		;;
	  *)
	  	# Failed to stop
		log_end_msg 1
		;;
	esac
	;;
  *)
	echo "Usage: $SCRIPTNAME {start|stop|status|restart|force-reload}" >&2
	exit 3
	;;
esac

:
